napari/utils/stubgen.py

```diff
@@ -83,6 +83,27 @@ def _iter_imports(hint) -> Iterator[str]:
         yield hint.__module__
 
 
+def _import_for_type_checking(module) -> ModuleType:
+    pre_mods = list(sys.modules)
+    if isinstance(module, str):
+        module = importlib.import_module(module)
+
+    # this is a hack:
```

Comment:
this is really quite ugly... but this seems to be the cost of using things like `__future__.annotations` and `TYPE_CHECKING` imports. we need to re-import modules with `typing.TYPE_CHECKING == True`, which only sometimes works.  but for now, it seems to be "good enough"